main.py

```diff
@@ -1,65 +1,340 @@
+    }),
+    ('state', {
+        'type': 'string',
+        'start_pos': 26,
+        'end_pos': 28
+    }),
+    ('plate', {
+        'type': 'string',
+        'start_pos': 28,
+        'end_pos': 36
+    }),
+    ('division', {
+        'type': 'string',
+        'start_pos': 36,
+        'end_pos': 40
+    }),
+    ('location', {
+        'type': 'string',
+        'start_pos': 40,
+        'end_pos': 75
+    }),
+    ('violation_desc', {
+        'type': 'string',
+        'start_pos': 75,
+        'end_pos': 95
+    }),
+    ('fine', {
+        'type': 'decimal',
+        'start_pos': 95,
+        'end_pos': 104
+    }),
+    ('issuing_agency', {
+        'type': 'string',
+        'start_pos': 104,
+        'end_pos': 112
+    }),
+    ('lat', {
+        'type': 'latlon',
+        'start_pos': 112,
+        'end_pos': 122
+    }),
+    ('lon', {
+        'type': 'latlon',
+        'start_pos': 122,
+        'end_pos': 132
+    }),
+    ('gps', {
+        'type': 'boolean'
+    }),
+    ('zip_code', {
+        'type': 'string'
+    })
+])
 
-    return row
+headers = list(fieldmap.keys())
 
-def create_date_time (row):
-    date_time = row['issue_date'] + ' ' + row['issue_time']
-    return datetime.strptime(date_time, '%m/%d/%Y %H:%M').strftime('%Y-%m-%d %H:%M:%S')
+def parse_lat_lon(value):
+    value = float(value)
+    if value == 0.0:
+        return None
+    return value
 
-def remove_ppa_division (division, row):
-    if row['issuing_agency'] == 'PPA':
-        return ''
+def parse_datetime(value):
+    if value[-5:] == '24:00':
+        value = value[:-5] + '23:59'
+    return datetime.strptime(value, '%m/%d/%Y%H:%M')
+
+typemap = {
+    'integer': int,
+    'string': lambda x: x.strip(),
+    'decimal': lambda x: Decimal(x.replace(',','')),
+    'numeric': float,
+    'datetime': parse_datetime,
+    'latlon': parse_lat_lon
+}
+
+def get_transform_row(latlon_input):
+    def transform_row(line):
+        row = OrderedDict()
+        for field in fieldmap.keys():
+            config = fieldmap[field]
+            if config['type'] == 'latlon' and not latlon_input:
+                row[field] = None
+            elif 'start_pos' in config and 'end_pos' in config:
+                row[field] = typemap[config['type']]\
+                                (line[0][config['start_pos']:config['end_pos']])
+            else:
+                row[field] = None
+
+        if not latlon_input:
+            row['lat'] = None
+            row['lon'] = None
+
+        if (row['division'] == '0000' or row['division'] == '00'):
+            row['division'] = None
+
+        if row['issuing_agency'] == 'RED LI':
+            row['issuing_agency'] = 'PRISON'
+
+        if row['issuing_agency'] != 'PPA':
+            row['division'] = None
+
+        return row.values()
+
+    return transform_row
+
+def anonymize(row):
+    global plates_counter, ticket_numbers_counter
+
+    out_row = list(row)
+
+    plate_key = row[2] + row[3]
```

Comment:
Is there a way to use a dictionary for row or constants for the indexes so the reference to `[2]` and `[3]` and `[0]` is easier to read in this function?

Reply:
Not with petl